go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go

```diff
@@ -0,0 +1,222 @@
+// +build windows
+
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os/exec"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types/current"
+
+	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/config"
+)
+
+// noNameNetNS - This is received for the infra container, in this case we
+// have to create the network endpoint and attach it to the infra container
+// containerNetNS - This is received for all the other containers from
+// the same pod, do not create the network endpoint in thi case
+const (
+	noNameNetNS    = "none"
+	containerNetNS = "container:"
+)
+
+// More details about the above constants can be found in the following PR:
+// https://github.com/kubernetes/kubernetes/pull/51063
+
+/**
+ * When the HNS Endpoint is created, it asks for the HNS Id in order to
+ * attach it to the desired network. This function finds the HNS Id of the
+ * network based on the gatewayIP. If more than one suitable network it's found,
+ * return an error asking to give the HNS Network Id in config.
+ */
+func getHNSIdFromConfigOrByGatewayIP(gatewayIP string) (string, error) {
+	if config.CNI.WinHNSNetworkId != "" {
+		logrus.Infof("Using HNS Network Id from config: %v", config.CNI.WinHNSNetworkId)
+		return config.CNI.WinHNSNetworkId, nil
+	}
+	hnsNetworkId := ""
+	hnsNetworks, err := hcsshim.HNSListNetworkRequest("GET", "", "")
+	if err != nil {
+		return "", err
+	}
+	for _, hnsNW := range hnsNetworks {
+		for _, hnsNWSubnet := range hnsNW.Subnets {
+			if strings.Compare(gatewayIP, hnsNWSubnet.GatewayAddress) == 0 {
+				if len(hnsNetworkId) == 0 {
+					hnsNetworkId = hnsNW.Id
+				} else {
+					return "", fmt.Errorf("Found more than one network suitable for containers, please specify win-hnsnetwork-id in config")
+				}
+			}
+		}
+	}
+	if len(hnsNetworkId) != 0 {
+		logrus.Infof("HNS Network Id found: %v", hnsNetworkId)
+		return hnsNetworkId, nil
+	}
+	return "", fmt.Errorf("Could not find any suitable network to attach the container")
+}
+
+/**
+ * createHNSEndpoint creates the HNS endpoint with the given configuration.
+ * On success it returns the created HNS endpoint.
+ */
+func createHNSEndpoint(hnsConfiguration *hcsshim.HNSEndpoint) (*hcsshim.HNSEndpoint, error) {
+	logrus.Infof("Creating HNS endpoint")
+	hnsConfigBytes, err := json.Marshal(hnsConfiguration)
+	if err != nil {
+		logrus.Errorf("Error creawting HNSEndpoint: %v", err)
```

Comment:
s/creawting/creating/g. Also I would change the log message with:
Could not create endpoint with HNSEndpoint error: %v